testsuite/MDAnalysisTests/lib/test_util.py

```diff
@@ -21,37 +21,39 @@
 #
 from __future__ import absolute_import, division
 
-from unittest import TestCase
-
-from six.moves import range, StringIO
-import six
-
-import numpy as np
-from numpy.testing import (assert_raises, assert_equal, assert_almost_equal,
-                           assert_array_almost_equal, assert_, assert_array_equal)
-
 import MDAnalysis as mda
-import MDAnalysis.lib.util as util
 import MDAnalysis.lib.mdamath as mdamath
-from MDAnalysis.lib.util import cached
+import MDAnalysis.lib.util as util
+import numpy as np
+import pytest
+import six
 from MDAnalysis.core.topologyattrs import Bonds
 from MDAnalysis.exceptions import NoDataError
-
-
+from MDAnalysis.lib.util import cached
 from MDAnalysisTests.datafiles import Make_Whole
+from numpy.testing import (assert_equal, assert_almost_equal,
+                           assert_array_almost_equal, assert_array_equal)
+from six.moves import range, StringIO
 
 
-def check_parse_residue(rstring, residue):
-    assert_equal(util.parse_residue(rstring), residue)
-
-
-def check_convert_aa_3to1(resname3, resname1):
-    assert_equal(util.convert_aa_code(resname3), resname1)
-
-
-def check_convert_aa_1to3(resname1, resname3_canonical):
-    assert_equal(util.convert_aa_code(resname1), resname3_canonical)
+def convert_aa_code_long_data():
+    # TODO: Find a clean solution for this.
+    aa = [
+        ('H', ('HIS', 'HISA', 'HISB', 'HSE', 'HSD', 'HIS1', 'HIS2', 'HIE', 'HID')),
+        ('K', ('LYS', 'LYSH', 'LYN')),
+        ('A', ('ALA',)),
+        ('D', ('ASP', 'ASPH', 'ASH')),
+        ('E', ('GLU', 'GLUH', 'GLH')),
+        ('N', ('ASN',)),
+        ('Q', ('GLN',)),
+        ('C', ('CYS', 'CYSH', 'CYS1', 'CYS2')),
+    ]
+    data = []
+    for resname1, strings in aa:
+        for resname3 in strings:
+            data.append((resname3, resname1))
```

Comment:
The point is not to make it a list comprehension, but a generator. To do that, replace this line by `yield (resname3, resname1)`, remove the `data` list, and remove the `return`. But as I said, it does not change much, I like the generator better but the list is fine too.